server/src/server.ts

```diff
@@ -1436,7 +1436,20 @@ async function onMessage(msg: p.Message) {
         params.textDocument.uri as utils.FileURI,
         params.textDocument.text,
       );
-      await sendUpdatedDiagnostics();
+
+      if (config.extensionConfiguration.incrementalTypechecking?.enable) {
```

Comment:
This is the other relevant change: we check the file with incremental compilation versus looking at potentially stale .compiler.log